OIPA/iati/parser/tests/test_2_03_parser.py

```diff
@@ -1368,82 +1392,95 @@ def setUp(self):
-            "version": '2.03',
-        }
-        self.iati_203_XML_file = E("iati-activities", **xml_file_attrs)
+                result_document_link_XML_element
 
-        dummy_source = synchroniser_factory.DatasetFactory.create(
-            name="dataset-2"
-        )
+            )
+        except RequiredFieldError as inst:
+            self.assertEqual(inst.field, 'iso-date')
+            self.assertEqual(inst.message, 'required attribute missing')
 
-        self.parser_203 = ParseManager(
-            dataset=dummy_source,
-            root=self.iati_203_XML_file,
-        ).get_parser()
+        # case 2 : 'iso-date' is not valid
+        result_document_link_attr = {
+            "url": 'www.google.com',
+            "format": 'something',
+            "iso-date": '25116600000'
 
-        self.parser_203.default_lang = "en"
+        }
+        result_document_link_XML_element = E(
+            'document-link',
+            **result_document_link_attr
+        )
 
-        assert (isinstance(self.parser_203, Parser_203))
+        try:
+            self.parser_203.iati_activities__iati_activity__result__document_link__document_date(  # NOQA: E501
+                result_document_link_XML_element
+            )
+        except RequiredFieldError as inst:
+            self.assertEqual(inst.field, 'iso-date')
+            self.assertEqual(inst.message, 'Unspecified or invalid. Date '
+                                           'should be of type xml:date.')
 
-        # Version
-        current_version = VersionFactory(code='2.03')
+        # case 3: 'iso-date' is not in correct range
+        result_document_link_attr = {
+            "url": 'www.google.com',
+            "format": 'something',
+            "iso-date": '18200915'
 
-        # Related objects:
-        self.activity = iati_factory.ActivityFactory.create(
-            iati_standard_version=current_version
+        }
+        result_document_link_XML_element = E(
+            'document-link',
+            **result_document_link_attr
         )
-        self.document_link = iati_factory.DocumentLinkFactory. \
-            create(url='http://someuri.com')
-
-        self.parser_203.register_model('Activity', self.activity)
-        self.parser_203.register_model('DocumentLink', self.document_link)
+        try:
+            self.parser_203.iati_activities__iati_activity__result__document_link__document_date(  # NOQA: E501
+                result_document_link_XML_element
+            )
+        except FieldValidationError as inst:
+            self.assertEqual(inst.field, 'iso-date')
+            self.assertEqual(inst.message, 'iso-date not of type xsd:date')
 
-    def test_result_document_link_document_date(selfs):
+        # all is good
         result_document_link_attr = {
             "url": 'www.google.com',
             "format": 'something',
-            # 'format_code will be got in the function
+            "iso-date": '20110506'  # this is correct iso-date
```

Comment:
Everything seems fine, but I think according to [https://www.w3schools.com/xml/schema_dtypes_date.asp](https://www.w3schools.com/xml/schema_dtypes_date.asp) and [http://reference.iatistandard.org/203/activity-standard/iati-activities/iati-activity/result/document-link/document-date/](http://reference.iatistandard.org/203/activity-standard/iati-activities/iati-activity/result/document-link/document-date/), this is not really a correct ISO date for XML :) The parser code works, but it's probably a bug in our `validate_date()` method in `iati/parser/iati_parser.py`, it should throw an error if the iso-date isn't according to XML standards.